packages/core/src/layout/Header/Header.tsx

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { Fragment, ReactNode, CSSProperties, FC } from 'react';
+import Helmet from 'react-helmet';
+import { Typography, Tooltip, makeStyles } from '@material-ui/core';
+import { Theme } from '../Page/Page';
+// import { Link } from 'shared/components';
+import { BackstageTheme } from '../../theme/theme';
+import Waves from './Waves';
+
+const useStyles = makeStyles<BackstageTheme>(theme => ({
+  header: {
+    gridArea: 'pageHeader',
+    padding: theme.spacing(3),
+    minHeight: 118,
+    width: '100%',
+    boxShadow: '0 0 8px 3px rgba(20, 20, 20, 0.3)',
+    position: 'relative',
+    zIndex: 100,
+    display: 'flex',
+    flexDirection: 'row',
+    flexWrap: 'wrap',
+    justifyContent: 'flex-end',
+    alignItems: 'center',
+  },
+  leftItemsBox: {
+    flex: '1 1 auto',
+  },
+  rightItemsBox: {
+    flex: '0 1 auto',
+    display: 'flex',
+    flexDirection: 'row',
+    flexWrap: 'wrap',
+    alignItems: 'center',
+    marginRight: theme.spacing(6),
+  },
+  title: {
+    color: theme.palette.bursts.fontColor,
+    lineHeight: '1.0em',
+    wordBreak: 'break-all',
+    fontSize: 'calc(24px + 6 * ((100vw - 320px) / 680))',
+    marginBottom: theme.spacing(1),
+  },
+  subtitle: {
+    color: 'rgba(255, 255, 255, 0.8)',
+    lineHeight: '1.0em',
+  },
+  type: {
+    textTransform: 'uppercase',
+    fontSize: 9,
+    opacity: 0.8,
+    marginBottom: 10,
+    color: theme.palette.bursts.fontColor,
+  },
+}));
+
+type HeaderStyles = ReturnType<typeof useStyles>;
+
+type Props = {
+  component?: ReactNode;
+  pageTitleOverride?: string;
+  style?: CSSProperties;
+  subtitle?: ReactNode;
+  title: ReactNode;
+  tooltip?: string;
+  type?: string;
+  typeLink?: string;
+};
+
+type TypeFragmentProps = {
+  classes: HeaderStyles;
```

Comment:
is this used as an optimization? in theory the fragment could just useStyles() itself. (same below)

Reply:
It just felt more natural for me. If styles are already available in parent component then I would pass them via props. Mainly because I treat `xFragment` components as internal part of `Header`. 

I would call it a micro-optimization here, therefore, I'm open for just using `useStyles` inside if required.

Reply:
I don't mind. Just checking!